lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -455,18 +455,27 @@ def do_3d_projection(self, renderer):
 
         fcs = (zalpha(self._facecolor3d, vzs) if self._depthshade else
                self._facecolor3d)
-        fcs = mcolors.to_rgba_array(fcs, self._alpha)
-        self.set_facecolors(fcs)
-
         ecs = (zalpha(self._edgecolor3d, vzs) if self._depthshade else
                self._edgecolor3d)
+
+        # Sort the points based on z coordinates
+        # Put vzs first to sort based on z value
+        z_markers = sorted(zip(vzs, np.column_stack([vxs, vys]), fcs, ecs),
+                           reverse=True)
+
+        [zzs, vps, fcs, ecs] = zip(*z_markers)
+
+        fcs = mcolors.to_rgba_array(fcs, self._alpha)
         ecs = mcolors.to_rgba_array(ecs, self._alpha)
+
         self.set_edgecolors(ecs)
-        PathCollection.set_offsets(self, np.column_stack([vxs, vys]))
+        self.set_facecolors(fcs)
+
+        PathCollection.set_offsets(self, vps)
 
-        if vzs.size > 0 :
+        if vzs.size > 0:
             return min(vzs)
```

Comment:
we don't need to do `min(vzs)` anymore. we now have the sorted `zzs`. Just return the last element (since you did a reversed sort).